libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -269,6 +269,56 @@ void GCS_MAVLINK::send_rangefinder_downward(const RangeFinder &rangefinder) cons
             rangefinder.voltage_mv_orient(ROTATION_PITCH_270) * 0.001f);
 }
 
+void GCS_MAVLINK::send_proximity(const AP_Proximity &proximity, uint16_t count_max) const
+{
+    // return immediately if no proximity sensor is present
+    if (proximity.get_status() == AP_Proximity::Proximity_NotConnected) {
+        return;
+    }
+
+    // return immediately if no tx buffer room to send messages
+    if (count_max == 0) {
+        return;
+    }
+
+    bool send_upwards = true;
+
+    // send horizontal distances
+    AP_Proximity::Proximity_Distance_Array dist_array;
+    const uint8_t horiz_count = MIN(count_max, PROXIMITY_MAX_DIRECTION);  // send at most PROXIMITY_MAX_DIRECTION horizontal distances
+    if (proximity.get_horizontal_distances(dist_array)) {
+        for (uint8_t i = 0; i < horiz_count; i++) {
+            mavlink_msg_distance_sensor_send(
```

Comment:
To complement what I said above, this should now use the CHECK_PAYLOAD_SIZE macro so it will return when there is no more space to send messages. This makes the `count_max` and `send_upward` not needed.